v6.0.0-alpha.12 (#7223)
mui/mui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 6.0.0-alpha.12

_Dec 16, 2022_

We'd like to offer a big thanks to the 6 contributors who made this release possible. Here are some highlights ✨:

> - 🚀 The `apiRef` is now available on `@mui/x-data-grid`

Reviewer: ```suggestion
- 🚀 The `apiRef` prop is now available in the `@mui/x-data-grid` package:
```

  ```tsx
  const apiRef = useGridApiRef();

  return (
    <DataGrid apiRef={apiRef} {...other} />
  )
  ```

> 

Reviewer: ```suggestion

  See [the documentation](https://next.mui.com/x/react-data-grid/api-object/) for more information.

```

- 🎁 The `DataGridPremium` now supports cell selection:

  ```tsx
  <DataGridPremium unstable_cellSelection />
  ```

  See [the documentation](https://next.mui.com/x/react-data-grid/selection/#cell-selection) for more information

- 🌍 Support the *Right To Left* orientation on the fields components
- 📚 Documentation improvements
- 🐞 Bugfixes

### `@mui/x-data-grid@v6.0.0-alpha.12` / `@mui/x-data-grid-pro@v6.0.0-alpha.12` / `@mui/x-data-grid-premium@v6.0.0-alpha.12`

#### Breaking changes

- The `showCellRightBorder` was renamed to `showCellVerticalBorder`
- The `showColumnRightBorder` was renamed to `showColumnVerticalBorder`

> - The `.MuiDataGrid-withBorder` CSS class was renamed to `.MuiDataGrid-withBorderColor` and it only sets `border-color` CSS property now.

Reviewer: Few more breaking changes from https://github.com/mui/mui-x/pull/6773
```suggestion
- The `.MuiDataGrid-withBorder` CSS class was renamed to `.MuiDataGrid-withBorderColor` and it only sets `border-color` CSS property now.
- The following undocumented properties from `apiRef` were removed: `footerRef`, `headerRef`, `columnHeadersElementRef`, `columnHeadersContainerElementRef`
- The `GridHeaderPlaceholder` component was removed.
- The `MAX_PAGE_SIZE` constant was removed.
- The `useGridScrollFn` hook was removed.
```

#### Changes

- [DataGrid] Support `apiRef` in Community package (#6773) @cherniavskii
- [DataGridPremium] Add support for cell selection (#6567) @m4theushw
- [DataGridPremium] Use separate cache for aggregation columns pre-processor (#7142) @m4theushw
- [DataGridPro] Fix missing border in right-pinned columns (#4197) @cherniavskii
- [DataGridPro] Fix wrong border color on skeleton cells (#7202) @cherniavskii

### `@mui/x-date-pickers@v6.0.0-alpha.12` / `@mui/x-date-pickers-pro@v6.0.0-alpha.12`

#### Changes

- [fields] Fix bug introduced by RTL in single input range fields (#7189) @alexfauquette
- [fields] Support RTL out of the box (#6715) @alexfauquette
- [pickers] Clean `autoFocus` behavior on fields and new pickers (#7153) @flaviendelangle
- [pickers] Fix label on the new range pickers (#7210) @flaviendelangle
- [pickers] Fix wrong component name on `StaticNextDateTime` (#7187) @flaviendelangle

### Docs

> - [docs] Add doc section about field placeholders localization (#7139) @flaviendelangle

Reviewer 2: ```suggestion
- [docs] Add docs section about field placeholders' localization (#7139) @flaviendelangle
```

- [docs] Create a `DemoGrid` component to unify demos with several components (#7057) @flaviendelangle
- [docs] Document aggregation selectors (#7148) @cherniavskii
- [docs] Fix 301 links to demo pages in API pages (#7197) @oliviertassinari
- [docs] Fix errors and warning in demos (#7209) @LukasTy
- [docs] Use `DemoContainer` and `DemoItem` on every picker demo (#7149) @flaviendelangle

### Core

> - [core] Fix broken test (#7179) @flaviendelangle

Reviewer 3: Nitpick, but IMHO this is probably irrelevant and does not provide much value being in changelog. 🤔

Reviewer 3: I'd still be up for removing this entry. 🙈 🤷

Author: Isn't it the goal of the `Core` section ?
List everything that does not impact our users ?

I mean, most of the `Core` PRs are like this one

Reviewer 3: Yes, they usually are, but in this case it does not change anything in particular, just fixes a test regression, that's why I'm saying that it is only sort of a noise in the changelog for the end user. 🤔

Author: But like most PRs listed in the `Core` section
I agree that this one is very pointless, and that some PRs in the `Core` section can have more impact 
For instance:

```
- [core] Add codemod to move l10n translation (#7027) @alexfauquette
```

Right now, from what I see we list all the PRs.
I don't think that removing just a few on a case by case is very interesting.
My point being, if we want to reduce the noise, maybe would could just stop listing any PR not impacting the end user.

Reviewer 3: Well, like I mentioned, it's just a nitpick and my opinion.
We could both exclude changes not impacting users or come up with more changelog categories.
Or just keep on including everything. 😃 
Any way works. 👍

Author: For the changelog, the core team is not putting them in `Core` and I think we should do like them:

```tsx
### `@mui/codemod@5.9.3`

- [codemod] Fix theme-spacing performance (#33691) @siriwatknp
- [codemod] Support @mui import for variant-prop (#33692) @siriwatknp
```

____

Apart from that, I'm interested in clarifying what `core` means and how it should be handled in the CHANGELOG.

I feel like it depends a lot on what we are putting inside `Core`.

Reviewer 3: Yup, indeed. It would definitely make sense putting `codemods` into a separate changelog section.
We could address that with the next codemod we do. 👍
